File: backend/Concord/Concord.Application/Models/ChannelResponse.cs
using System.Text.Json.Serialization;
using Concord.Application.Enums;

namespace Concord.Application.Models;

public class ChannelResponse
{
    [JsonPropertyName("Id")]
    public Guid Id { get; set; }

    [JsonPropertyName("ServerId")]
    public Guid ServerId { get; set; }

    [JsonPropertyName("Name")]
    public string? Name { get; set; }

    [JsonPropertyName("Type")]
    public ChannelType Type { get; set; }

    [JsonPropertyName("Position")]
    public int Position { get; set; }

    [JsonPropertyName("Created")]
    public DateTime Created { get; set; }

    [JsonPropertyName("UnreadCount")]
    public int UnreadCount { get; set; }
}

File: mobile/concord/lib/api/models/channel_response.dart
import 'enums.dart';
import 'json_utils.dart';

class ChannelResponse {
  const ChannelResponse({
    required this.id,
    required this.serverId,
    required this.name,
    required this.type,
    required this.created,
    required this.unreadCount,
  });

  factory ChannelResponse.fromJson(Map<String, dynamic> json) {
    return ChannelResponse(
      id: json.field('Id') as String,
      serverId: json.field('ServerId') as String,
      name: json.field('Name') as String,
      type: ChannelType.fromWire(json.field('Type')),
      created: parseDateTime(json.field('Created')),
      unreadCount: json.field('UnreadCount') as int,
    );
  }

  final String id;
  final String serverId;
  final String name;
  final ChannelType type;
  final DateTime created;

  final int unreadCount;
}
